main.go

```diff
@@ -83,6 +86,29 @@ func main() {
 		defer cancel()
 	}
 
+	// WSL needs the GPU disabled. See issue #10
+	// This method of checking for WSL has worked since mid 2016
+	// https://github.com/microsoft/WSL/issues/423#issuecomment-328526847
+	osReleaseFile, err := os.Open("/proc/sys/kernel/osrelease")
```

Comment:
Couple of comments:

- Can you check the value of `runtime.GOOS` in WSL ? I am guessing that will be linux. If so, I would like to check that first, and then try to open the file. That should reduce an unnecessary attempt to open a file for other platforms.
- You can directly call `ioutil.ReadFile` without trying to open the file first.
- Let's move this to a helper function and do all the stuff there. I am envisioning something like
`if runtime.GOOS == "linux" && isWSL() {...}`, where `isWSL()` does a ioutil.ReadFile and compares. If ReadFile returns an error, it can return false.
- You are creating an `NewExecAllocator` twice. There's no need to do that. We can add all the required opts and then do a final `NewExecAllocator`.